codec-http2/src/main/java/io/netty/handler/codec/http2/DelegatingDecompressorFrameListener.java

```diff
@@ -250,24 +244,12 @@ Http2Decompressor decompressor(Http2Stream stream) {
     }
 
     /**
-     * Release remaining content from the {@link EmbeddedChannel} and remove the decompressor
-     * from the {@link Http2Stream}.
+     * Release remaining content from the {@link EmbeddedChannel}.
      *
-     * @param stream The stream for which {@code decompressor} is the decompressor for
      * @param decompressor The decompressor for {@code stream}
      */
-    private void cleanup(Http2Stream stream, Http2Decompressor decompressor) {
-        final EmbeddedChannel channel = decompressor.decompressor();
-        if (channel.finish()) {
-            for (;;) {
-                final ByteBuf buf = channel.readInbound();
-                if (buf == null) {
-                    break;
-                }
-                buf.release();
-            }
-        }
-        decompressor = stream.removeProperty(propertyKey);
+    private static void cleanup(Http2Decompressor decompressor) {
+        decompressor.decompressor().finishAndReleaseAll();
```

Comment:
There was an ordering issue related to the stream being removed if an error occurred and attempting to return bytes to flow control. So if we removed the property we would not be able to return bytes to flow control in this condition. Ideally we would remove it, but I don't think it hurts anything to leave it in the property map.